test/test_layoutarray.jl

```diff
@@ -95,6 +96,9 @@ MemoryLayout(::Type{MyVector}) = DenseColumnMajor()
             S = Symmetric(MyMatrix(reshape(inv.(1:25),5,5) + 10I))
             @test cholesky(S).U ≈ cholesky!(deepcopy(S)).U
             @test cholesky(S,Val(true)).U ≈ cholesky(Matrix(S),Val(true)).U
+
+            B = Symmetric(BandedMatrix(reshape(inv.(1:25),5,5) + 10I))
```

Comment:
Use `MyMatrix` instead of `BandedMatrix` to avoid the issue raised above.

Reply:
The problem is that `MyMatrix` doesn't seem to be affected by issue #66, i.e. `cholesky(S)` is correctly inferred. I'll see if I can define a minimal type similar to `BandedMatrix`.

Reply:
I just realised that this is because `cholcopy(::MyMatrix)` creates a regular `Array`. The issue can be instead reproduced by calling the in-place `cholesky!(::MyMatrix)`, which avoids `cholcopy`. I'll update the tests accordingly.